File: leetcode/LC1/string/0392_is_subsequence.cpp
#include <string>
using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int ss = 0;
        int st = 0;
        
        // ss is pointer on s, st is pointer on t
        while(ss < s.size() && st < t.size()){
            if(s[ss] == t[st]){
                ss++;
                st++;
            }else{
                st++;
            }
        }
        return (ss == s.size());
        
        
    }
};

File: leetcode/LC/string/0392_is_subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        ss = 0
        st = 0
        
        while(st < len(t) and ss < len(s)):
            if t[st] == s[ss]:
                st += 1
                ss += 1
            else:
                st += 1
        if (ss == len(s)):
            return True
        return False
        